multimodal/src/autogluon/multimodal/utils.py

```diff
@@ -2236,3 +2244,529 @@ def compute_num_gpus(config_num_gpus: Union[int, float, List], strategy: str):
+    If input is numpy.ndarray, return is numpy.ndarray correspondingly.
+    """
+    if isinstance(xyxy, (tuple, list)):
+        if not len(xyxy) == 4:
+            raise IndexError("Bounding boxes must have 4 elements, given {}".format(len(xyxy)))
+        x1 = np.minimum(width - 1, np.maximum(0, xyxy[0]))
+        y1 = np.minimum(height - 1, np.maximum(0, xyxy[1]))
+        x2 = np.minimum(width - 1, np.maximum(0, xyxy[2]))
+        y2 = np.minimum(height - 1, np.maximum(0, xyxy[3]))
+        return x1, y1, x2, y2
+    elif isinstance(xyxy, np.ndarray):
+        if not xyxy.size % 4 == 0:
+            raise IndexError("Bounding boxes must have n * 4 elements, given {}".format(xyxy.shape))
+        x1 = np.minimum(width - 1, np.maximum(0, xyxy[:, 0]))
+        y1 = np.minimum(height - 1, np.maximum(0, xyxy[:, 1]))
+        x2 = np.minimum(width - 1, np.maximum(0, xyxy[:, 2]))
+        y2 = np.minimum(height - 1, np.maximum(0, xyxy[:, 3]))
+        return np.hstack((x1, y1, x2, y2))
+    else:
+        raise TypeError("Expect input xywh a list, tuple or numpy.ndarray, given {}".format(type(xyxy)))
+
+
+def _check_load_coco_bbox(
+    coco, entry: dict, min_object_area: Optional[Union[int, float]] = 0, use_crowd: Optional[bool] = False
+):
+    """
+    Check and load ground-truth labels. Modified from gluon cv.
+
+    Parameters
+    ----------
+    coco
+        The COCO data class.
+    entry
+        The image annotation entry.
+    min_object_area
+        Minimum object area to consider.
+    use_crowd
+        Use crowd or not.
+
+    Returns
+    -------
+    Valid objects to consider.
+    """
+    entry_id = entry["id"]
+    # fix pycocotools _isArrayLike which don't work for str in python3
+    entry_id = [entry_id] if not isinstance(entry_id, (list, tuple)) else entry_id
+    ann_ids = coco.getAnnIds(imgIds=entry_id, iscrowd=None)
+    objs = coco.loadAnns(ann_ids)
+    # check valid bboxes
+    valid_objs = []
+    width = entry["width"]
+    height = entry["height"]
+    for obj in objs:
+        if obj["area"] < min_object_area:
+            continue
+        if obj.get("ignore", 0) == 1:
+            continue
+        is_crowd = obj.get("iscrowd", 0)
+        if not use_crowd and is_crowd:
+            continue
+        # convert from (x, y, w, h) to (xmin, ymin, xmax, ymax) and clip bound
+        xmin, ymin, xmax, ymax = bbox_clip_xyxy(bbox_xywh_to_xyxy(obj["bbox"]), width, height)
+        # require non-zero box area
+        if obj["area"] > 0 and xmax > xmin and ymax > ymin:
+            cname = coco.loadCats(obj["category_id"])[0]["name"]
+            valid_objs.append(
+                {
+                    "xmin": xmin / width,
+                    "ymin": ymin / height,
+                    "xmax": xmax / width,
+                    "ymax": ymax / height,
+                    "class": cname,
+                    "is_crowd": is_crowd,
+                }
+            )
+    return valid_objs
+
+
+def from_coco(
```

Comment:
The utils file is becoming big. We can consider to refactor it in a next PR. We can create "utils" as a folder and put the detection functionalities as `utils/detection.py`.

Reply:
Will do it in next PR.